File: backend/app/Models/Report.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Report extends Model
{
    use HasFactory;

    public const STATUS_PENDING = 'pending';
    public const STATUS_IN_PROGRESS = 'in_progress';
    public const STATUS_RESOLVED = 'resolved';

    protected $fillable = [
        'user_id',
        'title',
        'description',
        'category',
        'latitude',
        'longitude',
        'image_path',
        'status',
    ];

    protected $appends = ['image_url'];

    protected function casts(): array
    {
        return [
            'latitude' => 'decimal:7',
            'longitude' => 'decimal:7',
        ];
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function getImageUrlAttribute(): ?string
    {
        if (! $this->image_path) {
            return null;
        }

        return asset('storage/'.$this->image_path);
    }
}

File: frontend/app/src/main/java/com/smartcity/app/Report.java
package com.smartcity.app;

import java.io.Serializable;

public class Report implements Serializable {
    public final long id;
    public final String title;
    public final String description;
    public final String category;
    public final String status;
    public final double latitude;
    public final double longitude;
    public final String imageUrl;

    public Report(long id, String title, String description, String category, String status,
                  double latitude, double longitude, String imageUrl) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.category = category;
        this.status = status;
        this.latitude = latitude;
        this.longitude = longitude;
        this.imageUrl = imageUrl;
    }
}
